Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1878,3 +1918,339 @@ def parse_date_string(date_string, date_format='%Y-%m-%dT%H:%M:%S'):
+    :param description: Why it's malicious
+
+    :return: A malicious DBot entry
+    :rtype: ``dict``
+    """
+    file_types = ('md5', 'sha1', 'sha256', 'crc32', 'sha512', 'ctph')
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower in ('ip', 'email'):
+        key = 'Address'
+    elif indicator_type_lower == 'url':
+        key = 'Data'
+    elif indicator_type_lower == 'domain':
+        key = 'Name'
+    elif indicator_type_lower == 'cve':
+        key = 'ID'
+    elif indicator_type_lower in file_types:
+        key = indicator_type_lower.upper()
+        indicator_type_lower = 'file'
+    else:
+        raise DemistoException('Wrong indicator type supplied: {}'.format(indicator_type))
+    entry = {
+        key: indicator,
+        'Malicious': {
+            'Vendor': vendor,
+            'Description': description
+        }
+    }
+    return {outputPaths[indicator_type_lower]: entry}
+
+
+class BaseClient(object):
+    def __init__(self,
+                 integration_name,
+                 integration_command_name,
+                 integration_context_name):
+        """Base Client for use in integrations.
+
+         :type integration_name: ``str``
+         :param integration_name:
+            Name of the integration as shown in the integration UI, for example: Microsoft Graph User.
+
+         :type integration_command_name: ``str``
+         :param integration_command_name:
+            Command names should be written in all lower-case letters,
+            and each word separated with a hyphen, for example: msgraph-user.
+
+         :type integration_context_name: ``str``
+         :param integration_context_name:
+            Context output names should be written in camel case, for example: MSGraphUser.
+
+         :return: No data returned
+         :rtype: ``None``
+         """
+        self._integration_name = str(integration_name)
+        self._integration_command_name = str(integration_command_name)
+        self._integration_context_name = str(integration_context_name)
+
+    @property
+    def integration_name(self):
+        """Property of integration name
+        return: Integration command name.
+        :rtype: ``str``
+        """
+        return self._integration_name
+
+    @property
+    def integration_context_name(self):
+        """Property of integration name context
+        return: Integration command name
+        :rtype: ``str``
+        """
+        return self._integration_context_name
+
+    @property
+    def integration_command_name(self):
+        """Property of integration name command
+        return: Integration command name
+        :rtype: ``str``
+        """
+        return self._integration_command_name
+
+
+# Will add only if 'requests' module imported
+if 'requests' in sys.modules:
+    class BaseHTTPClient(BaseClient):
+        def __init__(self, integration_name, integration_command_name, integration_context_name, server, base_suffix,
```

Comment:
maybe add the headers to here, with same logic as ok_codes
+ auth
and all the other stuff i missed